### DISCUSSION TITLE
Coerce findOne call to return an empty object

### DISCUSSION THREAD

**Discussion by crice88** [prisma/prisma#12468] [Q&A] [answered]
Here's my dilemma... I want the find methods to return an empty object instead of null when a record is not found. The use case being is that we are using Nestjs and it handles serialization for you so if `null` is returned from the query, it will return a status of 200 without a body. Since the requesting side is setup to read the json that is returned, it will throw an error if the body isn't serializable.

Trying to find a way to avoid having to include a check on each query to return an empty object if a null value is returned from the query. Here's an example:

```
return (await prisma.foo.findOne({ where: { id: 'abcd1234' } }) || {});
```

If Prisma were to return an empty object, I could exclude this check. Any ideas? I have several, though none are really that clean of a solution i/e create a facade, manually parse the responses (pull text then JSON.parse), just wrap everything.

**Comment by austincrim** [comment, marked as answer]
Hey @crice88 👋 ,

We do not have a built-in options for this. You could look into writing a [Prisma middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) to accomplish this, that way you at least have the logic consolidated into one spot.

Something like this could work:
```ts
const prisma = new PrismaClient()

prisma.$use(async (params, next) => {
  const result = await next(params)  // execute query with given parameters
  if (!result) return {}             // return an empty object if query returns null
})
```

Let me know if I can clarify that further!

**Reply by crice88** [reply]
Ohhhh that's a great idea!! Thanks @austincrim, didn't even think to go that route!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
